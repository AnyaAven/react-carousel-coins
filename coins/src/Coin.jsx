import React from "react";
import "./Coin.css";

const COIN_IMAGES = {
  heads: "./image1.png",
  tails: "./image2.png",
};

/**
 *
 */
function Coin({ sideShowing }) {
  //OPTION1
  //take sideShowing, and get file from coin_images
  // return image of coin in div

  return (
    <div>
      <img></img>
    </div>
  );
}
