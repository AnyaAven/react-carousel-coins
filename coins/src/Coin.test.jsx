import { describe, test, expect } from "vitest";
import React from "react";
import { render } from "@testing-library/react";
import Coin from "./Coin.jsx";

describe("Coin works", function () {

  test("renders without crashing", function () {
    render(<Coin sideShowing="heads" />);
  });

  test("coin shows heads", function () {
    const { container } = render(<Coin sideShowing="heads" />);
    const img = container.querySelector(".Coin-img");

    expect(img.getAttribute("src")).toContain("headsPenny.jpg")
    expect(img.getAttribute("alt")).toEqual("Coin heads")
  });

  test("coin shows tails", function () {
    const { container } = render(<Coin sideShowing="tails" />);
    const img = container.querySelector(".Coin-img");

    expect(img.getAttribute("src")).toContain("tailsPenny.jpg")
    expect(img.getAttribute("alt")).toEqual("Coin tails")
  });

});
