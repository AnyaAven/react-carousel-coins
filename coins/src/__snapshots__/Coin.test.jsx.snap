// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Coin works > matches snapshot 1`] = `
<div>
  <div
    class="Coin"
  >
    <img
      alt="Coin heads"
      class="Coin-img"
      src="/src/headsPenny.jpg"
    />
  </div>
</div>
`;
